tests/functional/xhr/deny-list.test.js

```diff
@@ -0,0 +1,74 @@
+const testDriver = require('../../../tools/jil')
+const fetchBrowsers = testDriver.Matcher.withFeature('fetch')
+
+testDriver.test('XHR ajax events deny bam server', function (t, browser, router) {
+  const ajaxPromise = router.expectXHRMetrics()
+  const rumPromise = router.expectRum()
+  const loadPromise = browser.safeGet(router.assetURL('instrumented.html', {
+    init: {
+      ajax: {
+        harvestTimeSeconds: 2,
+        enabled: true,
+        deny_list:[router.router.assetServer.host]
+      }
+    },
+    scriptString: `
+      const xhr = new XMLHttpRequest();
+      xhr.open('GET', 'http://example.com/');
+      xhr.send();
+    `
+  }))
+
+  Promise.all([ajaxPromise, loadPromise, rumPromise])
+    .then(([response]) => {
+      const newAjaxPromise = router.expectBeaconRequest(
```

Comment:
same here, can we use `expectXhrMetrics()` for `xhr` in errors payloads to ensure other payloads don't come through?

Reply:
I had to use expectBeaconRequest so I could set a custom timeout.